omnisend/includes/SDK/V1/class-event.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * Omnisend Client
+ *
+ * @package OmnisendClient
+ */
+
+namespace Omnisend\SDK\V1;
+
+use Omnisend\SDK\V1\Contact;
+use WP_Error;
+
+defined( 'ABSPATH' ) || die( 'no direct access' );
+
+/**
+ * Omnisend Event class. It's should be used with Omnisend Client.
+ *
+ */
+class Event {
+
+
+
+	private $contact          = null;
+	private $event_name       = null;
+	private $event_time       = null;
+	private $event_version    = null;
+	private $origin           = null;
+	private array $properties = array();
+
+	/**
+	 * Validate event properties.
+	 *
+	 * It ensures that all required properties are set
+	 *
+	 * @return WP_Error
+	 */
+	public function validate(): WP_Error {
+
+		$error = new WP_Error();
+		if ( $this->contact instanceof Contact ) {
+			$error->merge_from( $this->contact->validate() );
+		}
+
+		if ( $this->event_name == null ) {
+			$error->add( 'event_name', 'Is required.' );
+		}
+
+		if ( $this->event_name != null && ! is_string( $this->event_name ) ) {
+			$error->add( 'event_name', 'Not a string.' );
+		}
+
+		if ( $this->event_version != null && ! is_string( $this->event_version ) ) {
+			$error->add( 'event_version', 'Not a string.' );
+		}
+
+		if ( $this->origin != null && ! is_string( $this->origin ) ) {
+			$error->add( 'origin', 'Not a string.' );
+		}
+
+		if ( $this->event_time != null ) {
+			$dt = DateTime::createFromFormat( 'Y-m-d', $this->event_time );
+			if ( ! $dt ) {
+				$error->add( 'event_time', 'Not valid date format.' );
+			}
+		}
+
+		foreach ( $this->properties as $name => $value ) {
+			if ( ! is_string( $name ) ) {
+				$error->add( $name, 'Not a string.' );
+			}
+			if ( ! is_string( $value ) ) {
+				$error->add( $value, 'Not a string.' );
+			}
+		}
+
+		return $error;
+	}
+
+	/**
+	 * Sets event name.
+	 *
+	 * @param $event_name
+	 *
+	 * @return void
+	 */
+	public function set_event_name( $event_name ): void {
+		$this->event_name = $event_name;
+	}
+
+	/**
+	 * Sets event time. Time should be in RFC3339 (example 2011-01-01T00:00:00Z)
+	 *
+	 * @param $event_time
+	 *
+	 * @return void
+	 */
+	public function set_event_time( $event_time ): void {
+		$this->event_time = $event_time;
+	}
+
+	/**
+	 * Sets event version.
+	 *
+	 * @param $event_version
+	 *
+	 * @return void
+	 */
+	public function set_event_version( $event_version ): void {
+		$this->event_version = $event_version;
+	}
+
+
+	/**
+	 * Sets event origin. Default value is api
+	 *
+	 * @param $origin
+	 *
+	 * @return void
+	 */
+	public function set_origin( $origin ): void {
+		if ( $origin == null ) {
```

Comment:
This logic should be moved to `to_array` function. If user will not call this method validation will pass and user will get error from api.

Reply:
oh,ok. good point

Reply:
fixed